torch/_refs/__init__.py

```diff
@@ -805,6 +846,52 @@ def float_power(
     return prims.pow(a, b)
 
 
+def _floor_divide(
+    a: Union[TensorLikeType, NumberType], b: Union[TensorLikeType, NumberType]
+):
+    # Wrap scalars because some references only accept tensor arguments.
+    if isinstance(a, Number) and isinstance(b, Number):
+        a = scalar_tensor(a)
+        b = scalar_tensor(b)
+    elif isinstance(b, Number) and isinstance(a, Tensor):
+        b = scalar_tensor(b, dtype=a.dtype, device=a.device)
+    elif isinstance(a, Number) and isinstance(b, Tensor):
+        a = scalar_tensor(a, dtype=b.dtype, device=b.device)
+
+    # a // b = (a - remainder(a, b)) / b
+    mod = fmod(a, b)
+    div = true_divide(sub(a, mod), b)
+
+    # Ensure that the remainder has the same sign as denominator
+    # TODO: Should we use signbit reference here?
+    different_signed_inputs = bitwise_xor(lt(a, 0), lt(b, 0))
```

Comment:
Maybe we could use `signbit` reference here. It wouldn't change the function's behavior because `a` and `b` must be non-zero in this case.